src/Simplify.cpp

```diff
@@ -3908,6 +3924,68 @@ class Simplify : public IRMutator {
             IRMutator::visit(op);
         }
     }
+    template <typename T>
+    Expr hoist_slice_vector(Expr e) {
+        const T *op = e.as<T>();
+        internal_assert(op);
+
+        debug(4) << "Trying to hoist slice vector " << (Expr) op << "\n";
+
+        const Call *call_a = op->a.template as<Call>();
+        const Call *call_b = op->b.template as<Call>();
+
+        internal_assert(call_a && call_b &&
+                        call_a->is_intrinsic(Call::slice_vector) &&
+                        call_b->is_intrinsic(Call::slice_vector));
+
+        const int64_t *start_lane_a = as_const_int(call_a->args[1]);
+        const int64_t *start_lane_b = as_const_int(call_b->args[1]);
+        if (*start_lane_a != *start_lane_b) {
+            debug(4) << " ...unable to hoist - differing start lanes\n";
+            return e;
+        }
+
+        const int64_t *stride_a = as_const_int(call_a->args[2]);
+        const int64_t *stride_b = as_const_int(call_b->args[2]);
+        if (*stride_a != 1 || *stride_b != 1) {
+            debug(4) << " ...unable to hoist - differing strides\n";
+            return e;
+        }
+
+        const Call *concat_a = call_a->args[0].as<Call>();
+        const Call *concat_b = call_b->args[0].as<Call>();
+        if (!concat_a || !concat_b) {
+            debug(4) << " ...unable to hoist - both operands are not concat_vectors\n";
+            return e;
+        }
+
+        const std::vector<Expr> &slices_a = concat_a->args;
+        const std::vector<Expr> &slices_b = concat_b->args;
+        if (slices_a.size() != slices_b.size()) {
+            return e;
+        }
+
+        for (size_t i = 0; i < slices_a.size(); i++) {
+            if (slices_a[i].type() != slices_b[i].type()) {
+                debug(4) << " ...unable to hoist - type mismatch\n";
+                return e;
+            }
+        }
+
+        vector<Expr> new_slices;
+        for (size_t i = 0; i < slices_a.size(); i++) {
+            new_slices.push_back(T::make(slices_a[i], slices_b[i]));
+        }
+
+        Expr start_lane = call_a->args[1];
+        Expr result_lanes = call_a->args[3];
+        Expr concat_v = Call::make(concat_a->type, Call::concat_vectors, new_slices, Call::PureIntrinsic);
+        Expr ret_expr = Call::make(op->type, Call::slice_vector,
+                                   { concat_v, start_lane, 1 /*for now, we only deal with stride 1 */, result_lanes },
```

Comment:
Hmm, this seems related to my comment above... if the intention was to only deal with stride 1 vectors, a comment explaining why would be nice. But, I have a lot of code that uses stride 2 vectors that might benefit from this change, so it would be nice to try to support that.